portality/decorators.py

```diff
@@ -99,3 +100,43 @@ def decorated_view(*args, **kwargs):
 
         return decorated_view
     return decorator
+
+
+class CaughtTermException(Exception):
+    pass
+
+
+def _term_handler(signum, frame):
+    app.logger.warning("Harvester terminated with signal " + str(signum))
+    raise CaughtTermException
+
+
+def capture_sigterm(fn):
```

Comment:
this could be generalised further in future, right now it's harvester-specific.

Reply:
no fix required right now